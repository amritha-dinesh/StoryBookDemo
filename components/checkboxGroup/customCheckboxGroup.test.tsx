import React from "react";
import { render, fireEvent } from "@testing-library/react-native";
import CustomCheckboxGroup from "./customCheckboxGroup";

describe("CustomCheckboxGroup", () => {
  it("should match the snapshot", () => {
    const listOfCheckboxes = [
      { label: "Java", value: false },
      { label: "Kotlin", value: false },
    ];
    const { toJSON } = render(
      <CustomCheckboxGroup Checkboxes={listOfCheckboxes} />
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it("renders correctly", () => {
    const { getByTestId } = render(<CustomCheckboxGroup Checkboxes={[]} />);
    const customCheckboxGroup = getByTestId("custom-checkbox-group");
    expect(customCheckboxGroup).toBeDefined();
  });

  it("should render the correct number of checkboxes", () => {
    const listOfCheckboxes = [
      { label: "Java", value: false },
      { label: "Kotlin", value: false },
    ];
    const { getAllByTestId } = render(
      <CustomCheckboxGroup Checkboxes={listOfCheckboxes} />
    );
    const checkboxes = getAllByTestId("custom-checkbox");
    expect(checkboxes.length).toBe(listOfCheckboxes.length);
  });

  it("should call onClick function when a checkbox is clicked", () => {
    const onClickMock = jest.fn();
    const listOfCheckboxes = [{ label: "Java", value: false }];
    const { getByTestId } = render(
      <CustomCheckboxGroup
        Checkboxes={listOfCheckboxes}
        onClick={onClickMock}
      />
    );
    const checkbox = getByTestId("custom-checkbox");
    fireEvent.press(checkbox);
    expect(onClickMock).toHaveBeenCalledTimes(1);
    expect(onClickMock).toHaveBeenCalledWith(listOfCheckboxes[0], true, 0);
  });
});
