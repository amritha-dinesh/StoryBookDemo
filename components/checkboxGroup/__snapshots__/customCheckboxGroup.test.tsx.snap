// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomCheckboxGroup should match the snapshot 1`] = `
<View
  testID="custom-checkbox-group"
>
  <RCTSafeAreaView
    style={
      [
        {
          "alignItems": "center",
          "alignSelf": "flex-start",
        },
        {
          "flexDirection": "row",
        },
      ]
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": false,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
      testID="custom-checkbox"
    >
      <Image
        source={
          {
            "testUri": "../../../src/assets/unchecked.png",
          }
        }
        style={
          {
            "height": 40,
            "tintColor": "#9e9e9e",
            "width": 40,
          }
        }
      />
    </View>
    <Text
      style={
        {
          "color": "#000000",
          "fontFamily": undefined,
          "fontSize": undefined,
          "marginHorizontal": 10,
        }
      }
    >
      Java
    </Text>
  </RCTSafeAreaView>
  <RCTSafeAreaView
    style={
      [
        {
          "alignItems": "center",
          "alignSelf": "flex-start",
        },
        {
          "flexDirection": "row",
        },
      ]
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": false,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
      testID="custom-checkbox"
    >
      <Image
        source={
          {
            "testUri": "../../../src/assets/unchecked.png",
          }
        }
        style={
          {
            "height": 40,
            "tintColor": "#9e9e9e",
            "width": 40,
          }
        }
      />
    </View>
    <Text
      style={
        {
          "color": "#000000",
          "fontFamily": undefined,
          "fontSize": undefined,
          "marginHorizontal": 10,
        }
      }
    >
      Kotlin
    </Text>
  </RCTSafeAreaView>
</View>
`;
